### PR TITLE
D10 compatibility

### PR DESCRIPTION
**Pull request by andypost**
Ref https://www.drupal.org/node/3201242

### CODE REVIEW

**Comment by Sergey-Orlov** [comment]
added more fixes from https://www.drupal.org/project/rng/issues/3297977 for D10 compatibility

**Comment by andypost** [comment]
I think it ready to go but needs followup to update info files to support D11

File: composer.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "license": "GPL-2.0+",
   "require": {
     "drupal/courier": "^2.0@alpha",
-    "drupal/dynamic_entity_reference": "^1.0",
+    "drupal/dynamic_entity_reference": "^1 || ^3 || ^4",
```
Comment (andypost) [inline composer.json]:
```suggestion
    "drupal/dynamic_entity_reference": "^3 || ^4",
```

File: rng.info.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: RNG
 type: module
 description: 'Provides the ability to create and manage events and registrations.'
 package: RNG
-core_version_requirement: ^8.8.4 || ^9.0
+core_version_requirement: ^9.4 || ^10
```
Comment (andypost) [inline rng.info.yml]:
```suggestion
core_version_requirement: ^9.4 || ^10 || ^11
```

**Review by andypost — COMMENTED** [review]
Please fix DER on 3|4

File: composer.json (hunk 1 above)
Comment (Sergey-Orlov) [reply composer.json]: fixed

File: rng.info.yml (hunk 2 above)
Comment (Sergey-Orlov) [reply rng.info.yml]: updated core_version_requirement for submodules too

**Pull request merged**